app/dojo.py

```diff
@@ -0,0 +1,205 @@
+from random import choice
+from app.rooms import Room, Office, LivingSpace
+from app.person import Person, Fellow, Staff
+
+
+class Dojo():
+
+	def __init__(self):
+		self.all_rooms = []
+		self.offices = []
+		self.living_spaces = []
+		self.all_people = []
+		self.fellows = []
+		self.staffs = []
+		self.waiting_to_allocate_living = []
+		self.waiting_to_allocate_office = []
+
+	def create_room(self, room_type, room_names):
+		""" function to create a unique room space """
+		msg = ''
+		for name in room_names:
+			room_name = name.lower()
+			room_type = room_type.lower()
+
+			found = False
+			for room in self.all_rooms:
+				if room_name == room.room_name:
+					found = True
+			if found:
+				msg += "Room with name " + room_name + " already exists \n"
+			else:
+				if room_type == 'office':
+					room = Office(room_name)
+					self.offices.append(room)
+					self.all_rooms.append(room)
+					msg += "Office " + room_name + " added successfully \n"
+				elif room_type == 'living':
+					room = LivingSpace(room_name)
+					self.living_spaces.append(room)
+					self.all_rooms.append(room)
+					msg += "Living space " + room_name + " added successfully\n"
+				else:
+					msg += "Wrong room type \n"
+		return msg
+
+
+	def add_person(self, first_name, last_name, category, accomodation="N"):
+		""" function to add a person """
+		if isinstance(first_name, str) and isinstance(last_name, str):
+			if category.lower() == "fellow":
+				person = Fellow(first_name, last_name, accomodation)
+				person.id = int(len(self.all_people)+1)
+				self.fellows.append(person)
+				self.all_people.append(person)
+				self.waiting_to_allocate_office.append(person)
```

Comment:
Jus as you appending to `waiting_to_allocate_living` next to the script that allocates to the living I think you should append to `waiting_to_allocate_office` next to its related script.
 
Assuming something goes wrong between you appending and actually running your allocation scripts, there will also be inconsistent data to think about at that point. Your list of `waiting_to_allocate_office` will show more persons than it should leading to odd and hard to debug behaviour elsewhere.